### PR TITLE
Cleans up unused code

### PR DESCRIPTION
**Pull request by javiercn**
Adds a class and new methods to consume diagnostics from the tool and capture more detailed information on errors.
* dotnet-dev certs and the xplat library will be updated to consume the new methods.
* After that, the methods will be renamed to the old names and everything will get updated again.
* Cleans up dead code

### CODE REVIEW

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs [hunk 1]
```diff
@@ -492,18 +529,23 @@ private static void RemoveCertificateFromUserStore(X509Certificate2 certificate)
...
+            diagnostics?.Debug($"Trying to remove certificate with thumbprint '{certificate.Thumbprint}' from certificate store '{StoreLocation.CurrentUser}\\{StoreName.Root}'.");
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
                 using (var store = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
                 {
                     store.Open(OpenFlags.ReadWrite);
                     var matching = store.Certificates
                         .OfType<X509Certificate2>()
-                        .Single(c => c.SerialNumber == certificate.SerialNumber);
+                        .SingleOrDefault(c => c.SerialNumber == certificate.SerialNumber);
```
Comment (javiercn) [inline shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: This was a bug that surfaced when the certificate was not trusted.

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs [hunk 2]
```diff
@@ -362,13 +366,21 @@ private void TrustCertificateOnMac(X509Certificate2 publicCertificate)
...
+                if (existing.Count > 0)
+                {
+                    diagnostics?.Debug("Certificate already trusted. Skipping trust step.");
+                    DisposeCertificates(existing.OfType<X509Certificate2>());
+                    return;
+                }
+
                 try
                 {
                     store.Add(publicCertificate);
```
Comment (pranavkm) [inline shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: Does this need logging?

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs [hunk 3]
```diff
@@ -56,8 +51,10 @@ internal class CertificateManager
             StoreName storeName,
             StoreLocation location,
             bool isValid,
-            bool requireExportable = true)
+            bool requireExportable = true,
+            DiagnosticInformation diagnostics = null)
```
Comment (davidfowl) [inline shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: What is this?

Comment (pranavkm) [reply shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: Home brew logging

**Pull request merged**